### ISSUE TITLE
Default component prefixes for automatic inner declarations

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/ModelicaSpecification#1749]
**Reported by hansolsson on 17 Jul 2015 07:47 UTC**
https://trac.modelica.org/Modelica/ticket/1551#comment:12
added automatic inner declarations (which is good) and also stated:

The annotations defined in 17.6 does not affect this process, other than that: 
* defaultComponentPrefixes affects what prefixes the automatically generated declaration has 
* missingInnerMessage can be used for the diagnostic (and possibly error messages) 

The missingInnerMessage-part is ok, but I would propose to remove the defaultComponentPrefixes-part as noted in https://trac.modelica.org/Modelica/ticket/1551#comment:22

Basically we should always make the automatic "inner" component "inner" (!) - even without defaultComponentPrefixes="inner", and never make it "outer" (since that would require a new "inner" component etc) - even if it had defaultComponentPrefixes="outer".
I assume someone could use "parameter" as well, but the idea seems farfetched and I haven't seen any actual examples.

Or basically: the defaultComponentPrefixes-part causes more problems than it is worth, so just remove that part.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1749

**Comment by modelica-trac-importer** [comment]
**Comment by stefanv on 17 Jul 2015 14:16 UTC**
I think `defaultComponentPrefixes` should have **no** effect on this. My understanding of that annotation is that it is for the GUI to use when the user drags a component onto the canvas, and has nothing do with components introduced during flattening. It is, after all, described in a section entitled, _Annotations for the Graphical User Interface_.

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 17 Jul 2015 14:23 UTC**
Replying to [comment:1 stefanv]:
> I think `defaultComponentPrefixes` should have **no** effect on this. My understanding of that annotation is that it is for the GUI to use when the user drags a component onto the canvas, and has nothing do with components introduced during flattening. It is, after all, described in a section entitled, _Annotations for the Graphical User Interface_.

Agreed, that was also my conclusion.
I don't know why we previously decided that automatic inner should use that annotation.

**Comment by modelica-trac-importer** [comment]
**Comment by jmattsson on 30 Sep 2015 08:13 UTC**
Replying to [comment:2 hansolsson]:
> I don't know why we previously decided that automatic inner should use that annotation.

I think it was added as an "oh, by the way" that wasn't thought through enough. I agree with removing it from the list.

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 2 Oct 2015 20:51 UTC**
Ticket retargeted after milestone closed

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 1 Apr 2016 13:18 UTC**
There is a clear proposal to remove the defaultComponentPrefixes-part from automatic inner (and general agreement about it).

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 21 Jun 2016 11:31 UTC**
 Language group: In agreement by acclamation.

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 23 Jun 2016 11:40 UTC**
Scheduled to be added to spec.

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 23 Sep 2016 12:20 UTC**
Resolved in r9489
